src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -4160,7 +4160,24 @@ internal void TraceVariableSet(string varName, object value)
             // because 'ToStringParser' would iterate through the enumerator to get the individual elements, which will
             // make irreversible changes to the enumerator.
             bool isValueAnIEnumerator = PSObject.Base(value) is IEnumerator;
-            string valAsString = isValueAnIEnumerator ? typeof(IEnumerator).Name : PSObject.ToStringParser(_context, value);
+            string valAsString = string.Empty;
+            if (isValueAnIEnumerator)
+            {
+                valAsString = typeof(IEnumerator).Name;
+            }
+            else
+            {
+                try
+                {
+                    valAsString = PSObject.ToStringParser(_context, value);
+                }
+                catch
+                {
+                    // ToStringParser() can throw if the value is already disposed.
+                    valAsString = "null";
```

Comment:
This isn't really accurate -- the value isn't null, and our assumption is that this code path is reached in the event the object has been disposed (even though we catch any exception here).

Is `"null"` a string with a special meaning for us, or would something like `"<disposed>"` work just as well?